File: vue3_apponintments/src/api/doctor.js
import request from '@/utils/request.js'
//修改医生数据
export const UpdateDoctors = (data) => {
    return request.post('/doctor/update',data )
}
//获取医生列表
export const GetDoctors = (data) => {
    return request.post('/admin/alldoctor',data )
}
//删除医生
export const DeleteDoctors = (data) => {
    return request.post('/doctor/delete',data )
}
//添加医生
export const AddDoctors = (data) => {
    return request.post('/doctor/add',data )
}

//获取对应患者列表
export const Gtepatient = (data) => {
    return request.post('/doctor/gtepatient',data )
}

//预约已完成
export const Setappointment = (data) => {
    return request.post('/doctor/setappointment',data )
}

//获取请假表
export const Getapplication = (data) => {
    return request.post('/doctor/application/get',data )
}

//进行请假
export const Addapplication = (data) => {
    return request.post('/doctor/application/add',data )
}


//请假单的修改
export const Setapplication = (data) => {
    return request.post('/doctor/application/set',data )
}


//请假单的修改
export const Gtedoctorschedule = (data) => {
    return request.post('/doctor/gtedoctorschedule',data )
}

File: gin_vue(5)/gin_vue/ruoters/doctor.go
package ruoters

import (
	"gin_vue/config"
	"gin_vue/controllers"
	"github.com/gin-gonic/gin"
)

func DoctorRuotersInit(r *gin.Engine) {
	patientGroup := r.Group("/doctor")
	{
		patientController := controllers.DoctorController{}
		patientGroup.POST("/add", patientController.AddDoctor)                                                        //医生的创建
		patientGroup.POST("/register", patientController.Register)                                                    //医生登录
		patientGroup.POST("/update", config.ValidateRegisteredClaims, patientController.Update)                       //修改
		patientGroup.POST("/delete", config.ValidateRegisteredClaims, patientController.Delete)                       //删除
		patientGroup.POST("/gtedoctorschedule", config.ValidateRegisteredClaims, patientController.GteDoctorsChedule) //查看个人排班
		patientGroup.POST("/gtepatient", patientController.Gtepatient)                                                //查看个人患者信息
		patientGroup.POST("/setappointment", patientController.SetAppointment)
	}

	applicationGroup := r.Group("/doctor/application")
	{
		applicationController := controllers.ApplicationController{}
		applicationGroup.POST("/add", applicationController.AddApplication) //申请签到（医生）
		applicationGroup.POST("/get", applicationController.GetApplication) //获取申请表（医生）（管理员）
		applicationGroup.POST("/set", applicationController.SetApplication) //对申请表的操作（修改）（管理员）
	}
}
